### PR TITLE
Jl/caip mutlichain migrate core convert create session ts

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Convert wallet_createSession to ts

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28020?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by shanejonas — APPROVED** [review]

File: app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts [hunk 1]
```diff
@@ -97,12 +180,14 @@ export async function walletCreateSessionHandler(req, res, _next, end, hooks) {
     const supportedEthAccounts = getEthAccounts({
       requiredScopes: supportedRequiredScopes,
       optionalScopes: supportedOptionalScopes,
+      isMultichainOrigin: true,
```
Comment (adonesky1) [inline app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts:183]: this looks like a functional change?

Comment (adonesky1) [reply app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts:183]: I mean it looks right but wondering why it wasn't causing issues previously?

Comment (jiexi) [reply app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts:183]: yeah let's do that. I'll update core

Comment (jiexi) [reply app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts:183]: https://github.com/MetaMask/core/pull/4835

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**